src/frontend/screens/Settings/MapSettings/OfflineAreas.tsx

```diff
@@ -0,0 +1,173 @@
+import { ScrollView, StyleSheet } from "react-native";
+import { NavigationStackScreenComponent } from "react-navigation-stack";
+
+import { RED } from "../../../lib/styles";
+import {
+  BottomSheetContent,
+  BottomSheetModal,
+  useBottomSheetModal,
+} from "../../../sharedComponents/BottomSheetModal";
+import Button from "../../../sharedComponents/Button";
+import HeaderTitle from "../../../sharedComponents/HeaderTitle";
+import { ErrorIcon } from "../../../sharedComponents/icons";
+import Loading from "../../../sharedComponents/Loading";
+import { OfflineAreaCard } from "../../../sharedComponents/OfflineAreaCard";
+
+const m = defineMessages({
+  title: {
+    id: "screens.Settings.MapSettings.OfflineArea",
+    defaultMessage: "Offline Areas",
+  },
+  removeMap: {
+    id: "screens.Settings.MapSettings.removeMap",
+    defaultMessage: "Remove Map",
+  },
+  cancel: {
+    id: "screens.Settings.MapSettings.cancel",
+    defaultMessage: "Cancel",
+  },
+  clearDiagrams: {
+    id: "screens.Settings.MapSettings.clearDiagrams",
+    defaultMessage: "Are you sure you want to clear diagrams?",
+  },
+  subtitle: {
+    id: "screens.Settings.MapSettings.subtitle",
+    defaultMessage:
+      "This map and offline areas attached to it will be deleted. This cannot be undone",
+  },
+});
+
+interface OfflineArea {
+  id: string;
+  title: string;
+  zoomLevel: number;
+}
+
+export const OfflineAreas: NavigationStackScreenComponent = ({
+  navigation,
+}) => {
+  const bgMapId = React.useRef("");
+  const { formatMessage: t } = useIntl();
+
+  const { closeSheet, isOpen, openSheet, sheetRef } = useBottomSheetModal({
+    openOnMount: false,
+  });
+
+  const [offlineAreaList, setOfflineAreaList] = React.useState<OfflineArea[]>();
+
+  const { getParam } = navigation;
+
+  React.useEffect(() => {
+    bgMapId.current = getParam("mapId", "");
+
+    //To Do Api call to get offline areas
+    function getAllOfflineAreas(mapId: string): OfflineArea[] {
+      return [
+        {
+          id: "1",
+          title: "Offline Area 1",
+          zoomLevel: 12,
+        },
+        {
+          id: "2",
+          title: "Offline Area 2",
+          zoomLevel: 7,
+        },
+      ];
+    }
+
+    setOfflineAreaList(getAllOfflineAreas(bgMapId.current));
+  }, [getParam]);
+
+  return (
+    <React.Fragment>
+      <ScrollView style={[styles.container]}>
+        {offlineAreaList === undefined ? (
+          <Loading />
+        ) : (
+          offlineAreaList.map((offlineArea: OfflineArea, ind) => (
+            <OfflineAreaCard
+              key={offlineArea.id}
+              title={offlineArea.title}
+              zoomLevel={offlineArea.zoomLevel}
+              position={ind}
+            />
+          ))
+        )}
+      </ScrollView>
+
+      <Button style={styles.button} onPress={openSheet}>
+        {t(m.removeMap)}
+      </Button>
+
+      <DeleteAreaModal closeSheet={closeSheet} sheetRef={sheetRef} />
```

Comment:
Same comment as [this](https://github.com/digidem/mapeo-mobile/pull/946/files#r845363208). Either inline the component here or update the component to use `forwardRef`